### PR TITLE
UIU-3219 - navigate to pre patron registrations page and  render list

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
[UIU-3219](https://folio-org.atlassian.net/browse/UIU-3219) - Add the 'Search patron preregistration records’ action in the Action menu of User app
[UIU-3220](https://folio-org.atlassian.net/browse/UIU-3220) - Patron preregistration record results page - basic layout
[UIU-3221](https://folio-org.atlassian.net/browse/UIU-3221) - Add ‘Users: Can view patron preregistration data’ permission
[UIU-3222]( https://folio-org.atlassian.net/browse/UIU-3222) - Search patron preregistration data and display search results

## Approach
This feature is being developed on critical timelines. 
I am on vacation of rest of the week. These changes are critical for further stories implementations that would begin Monday, next week.
So, I would request the reviewers(@folio-org/fe-tl-reviewers ) to prioritize this PR review; if any non-critical observations, please write your comments and provide your approvals and we (@gurleenkaurbp and I) shall take care of them - probably in the scope of a tech debt or whatever suites the best.

**Also please note that coverage would be improved in the scope of https://folio-org.atlassian.net/browse/UIU-3226**

Update 1: Maximum coverage has been achieved. The above mentioned tech debt ticket will be closed.
Update 2: @zburke As per your review comment, implementation of https://folio-org.atlassian.net/browse/UIU-3221 is also included in this PR, in the interest of time. Please note that I have created the new permission on snapshot via postman. 
Also, I have attached the new screencast. Could you please re-review it now?

## Screencast
chrome_Ik4Nwvenlo

After update2, as explained above
msedge_OztpCw8H9I

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by Dmytro-Melnyshyn — APPROVED** [review]

File: src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js
Comment (zburke) [inline src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js:22]: `<FormattedMessage id="stripes-components.addNew" />` will give you a `New` label. `stripes-core.button.new` gives you `+ New`.

**Review by zburke — CHANGES_REQUESTED** [review]
What interface provides the `/staging-users` endpoint? At the very least, it needs to be added to `package.json` under `optionalOkapiInterfaces` and the work here fenced off by `<IfInterface>`. Alternatively, bump the major version and add that interface to `okapiInterfaces`.

File: src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js
Comment (gurleenkaurbp) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js:22]: Hello @zburke ,
Users interface provides this end point, the BE version will be increased and merged to master soon.
CC: @Terala-Priyanka

Comment (zburke) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js:22]: Then you have two options:
1. add `"staging-users": "1.0"` to `package.json` under `okapiInterfaces` and bump the major version of this module.
2. add `"staging-users": "1.0"` to `package.json` under `optionalOkapiInterfaces` and fence off all code that uses it behind `<IfInterface>` conditionals, leaving the version of this module as-is.

Comment (Terala-Priyanka) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js:22]: @zburke 
we will add "staging-users": "1.0" to package.json under okapiInterfaces (once it is available), mark as breaking in change log, but will retain the major version as is, because it has been bumped up already in the scope of Ramsons.

Comment (zburke) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js:22]: Additionally, the description for [UIU-3219](https://folio-org.atlassian.net/browse/UIU-3219) states "As a LoC registration staff..." Stripes cannot make this assertion; it has no notion of a specific tenant. This ticket, and therefore this PR, must not be merged as-is because they make an assertion that cannot be evaluated.

The closest we can come would be to wrap this feature behind a new permission set, but this implementation uses an existing one, `ui-users.create`.

Comment (Terala-Priyanka) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js]: we were trying to achieve some parallellism in the interest of time.

Comment (gurleenkaurbp) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js]: Hello @zburke , 
Can you please review this PR on urgent basis we are running against time atm :)
Thanks for your immediate attention.

Comment (zburke) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js]: Sorry, @gurleenkaurbp, but I cannot approve this as is. I disagree with how the story was written and I disagree with how this PR implements it. This change will show the patron pre-registration list to users with "create" permission; that is wrong. I understand that you attempted to break up this story by putting the new permission set in a separate story; that is wrong. 

To do this work in a way that avoids impacting existing implementation, this new feature must be implemented behind a conditional such as a feature flag, an optional interface dependency, or a permission set. None of those conditionals are implemented here. I understand the permission set will be added in UIU-3221. Fine, then block this ticket on that one and implement that story first. 

I'm sorry if this seems petty but IMO this work is not being done safely or correctly. I will dismiss my "request-changes" review but I will not approve this as is.

File: src/index.js
Comment (usavkov-epam) [inline src/index.js:218]: You protect the action but not the route

https://github.com/folio-org/ui-users/pull/2755/files#diff-801062542feb64887040b0917e0cb3fcc76055e2c6a0c979c0694121f21001d8R15

Comment (Terala-Priyanka) [reply src/index.js:218]: Yes, instead the search button on the new route is disabled in the absence of permission. This is agreed by our PO.

**Review by zburke — CHANGES_REQUESTED** [review]
@usavkov-epam's observation that we protect the button but not the route, meaning that even without permission you can still get to the route if you happen to know about it, is correct though. This bug must be fixed before you merge. 

My other quibbles are minor. It would be nice to get them cleaned up, but they are not blockers.

File: src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js
Comment (gurleenkaurbp) [reply src/views/PatronsPreRegistrationListContainer/PatronsPreRegistrationList.js]: Hello @zburke 
Please take a look at the pr's I hope they suffice now for the approval

**Comment by Terala-Priyanka** [comment]
> @usavkov-epam's observation that we protect the button but not the route, meaning that even without permission you can still get to the route if you happen to know about it, is correct though. This bug must be fixed before you merge.
> 
> My other quibbles are minor. It would be nice to get them cleaned up, but they are not blockers.

Hey @zburke 
I have fixed all your comments. Please re-review

**Review by zburke — APPROVED** [review]

**Pull request merged**
